### ISSUE TITLE
no-invalid-interactive triggers on <summary> tag

### ISSUE THREAD

**Issue by wuron** [ember-template-lint/ember-template-lint#1625]
Adding `{{on}}` modifier to `<summary>` tag triggers `no-invalid-interactive` rule. This seems incorrect, given that summary is a clickable element that has implicit ARIA role "button".

**Comment by zelaznik** [comment]
@wuron I'm not part of the core team, but I just saw this issue on the GitHub page, and I think you're right.

Before the rule `no-invalid-interactive` existed, I wrote my own custom rule that does the more or less the same thing.  This is a hopefully exhaustive list of CSS selectors that match to interactive elements.  I copied these selectors from a course on accessibility from Frontend Masters.

https://frontendmasters.com/courses/web-accessibility/exercise-2-solution/
https://github.com/jkup/focusable/blob/master/index.js

```js
const VALID_INTERACTIVE_SELECTORS = [
  'a[href]',
  'area[href]',
  'input:not([disabled]):not([type="hidden"])',
  'select:not([disabled])',
  'textarea:not([disabled])',
  'button:not([disabled])',
  'iframe',
  'object',
  'embed',
  '[tabindex="0"]',
  '[contenteditable]',
  'audio[controls]',
  'video[controls]',
  'summary',
  '[tabindex^="0"]',
  '[tabindex^="1"]',
  '[tabindex^="2"]',
  '[tabindex^="3"]',
  '[tabindex^="4"]',
  '[tabindex^="5"]',
  '[tabindex^="6"]',
  '[tabindex^="7"]',
  '[tabindex^="8"]',
  '[tabindex^="9"]',
];
```

I'm looking through the helper function that determines if an element is interactive.

https://github.com/ember-template-lint/ember-template-lint/blob/v3.0.0-beta.3/lib/helpers/is-interactive-element.js

We're missing `<summary>`, `<area>`.  I think we're also missing tags such as `<p contenteditable></p>`.  Such a tag is an affordance theory disaster, but it is technically valid.

I can submit a PR later today that updates the helper function `is-interactive-element`.

**Comment by zelaznik** [comment]
I tabled the issues that I raised that were unrelated to `<details>` and `<summary>`, but here's the PR that fixes this bug:

https://github.com/ember-template-lint/ember-template-lint/pull/1743

**Comment by MelSumner** [comment]
Here's the link to the spec for details, outlining that it is an interactive element: https://html.spec.whatwg.org/multipage/interactive-elements.html#the-details-element

Unfortunately, the details element is one of those things that could be awesome but wasn't implemented very well. :(

An interesting article on the subject: https://daverupert.com/2019/12/why-details-is-not-an-accordion/

**Comment by zelaznik** [comment]
Thanks for the reference.  I'll heed that advice going forward.

Looking through the guidelines, what would you say if I modified the PR so that both `<details>` and `<summary>` tags were listed as interactive elements?  That seems like @wuron's original issue, that he put a listener on the `<summary>` tag and it raised an error when it shouldn't have.

**Comment by lifeart** [comment]
Should be fixed in https://github.com/ember-template-lint/ember-template-lint/pull/3203
